### DISCUSSION TITLE
Upload a video in Ternoa's IPFS

### DISCUSSION THREAD

**Discussion by brucelane** [capsule-corp-ternoa/ternoa-js#100] [General]
Hi,

Following the discussion #62 , the next step in my app development is to upload a video to Ternoa's IPFS.
Does it work the same way?
Because there would be a thumbnail image AND the video too...
Thanks.

**Comment by ipapandinas** [comment]
This is a good question, thank you for asking.

You can upload a video file on Ternoa's IPFS node, the process remains the same as for an image and it was explained [here](https://github.com/capsule-corp-ternoa/ternoa-js/discussions/62#discussioncomment-3556905) on the previous discussion #62.

Regarding the thumbnail image there is no standardised solution, let me suggest two ideas:

1. Extend metadadas

The minimal metadata json file can be extended with a thumbnail field that aims to be the thumbnail image IPFS hash. You can then handle it on the application side.

2. Use secret NFTs (phase 4)

Secret NFTs will be released on phase 4 and this will open the door to new designs. The thumbnail image could be the NFT public cover and the secret NFT part the video.
This solution can be implemented later at the end of phase 4.

**Comment by brucelane** [comment]
Hi,
This json works:

````
{
  "title": "Ternoa Builder Badge",
  "description": "This badge is the gateway to the Ternoa Builders Experience.",
  "image": "Qm...hash of the thumbnail",
  "properties": {
    "media": {
      "hash": "Qm...hash of the video",
      "name": "Glass_Cube.mp4",
      "size": 6504265,
      "type": "video/mp4"
    }
  }
}
```
